libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -290,7 +290,9 @@ class FinalChainImpl final : public FinalChain {
       trxs_trie[i_rlp] = trx->rlp();
       auto const& receipt = receipts[i];
       receipts_trie[i_rlp] = util::rlp_enc(rlp_strm, receipt);
-      db_->insert(batch, DB::Columns::final_chain_receipt_by_trx_hash, trx->getHash(), rlp_strm.out());
+      if (!kLightNode) {
+        db_->insert(batch, DB::Columns::final_chain_receipt_by_trx_hash, trx->getHash(), rlp_strm.out());
+      }
```

Comment:
The main reason why we keep one week of history in period_data is because we want other nodes to be able to sync last week of data in between snapshots. final_chain_receipt_by_trx_hash is basically just used for rpc calls and the idea is that light node should not be used for rpc at all.

Another way to handle this is for final_chain_receipt_by_trx_hash is not to keep full week of history but maybe just last 1000 blocks

Reply:
I think light node should provide full functional rpc for period that we keep